packages/site/src/pages/index.tsx

```diff
@@ -76,35 +77,40 @@ const Index = () => {
     getState().catch((error) => console.error(error));
   }, [state.installedSnap]);
 
-  const createAccount = async () => {
-    const newAccount = await client.createAccount();
+  const syncAccounts = async () => {
     const accounts = await client.listAccounts();
     setSnapState({
       ...snapState,
       accounts,
     });
+  };
+
+  const createAccount = async () => {
+    const newAccount = await client.createAccount();
+    await syncAccounts();
     return newAccount;
   };
 
   const importAccount = async () => {
     const newAccount = await client.createAccount({
       privateKey: privateKey as string,
     });
-    const accounts = await client.listAccounts();
-    setSnapState({
-      ...snapState,
-      accounts,
-    });
+    await syncAccounts();
     return newAccount;
   };
 
   const deleteAccount = async () => {
     await client.deleteAccount(accountId as string);
-    const accounts = await client.listAccounts();
-    setSnapState({
-      ...snapState,
-      accounts,
-    });
+    await syncAccounts();
+  };
+
+  const updateAccount = async () => {
+    if (!accountObject) {
+      return;
+    }
+    const account = JSON.parse(accountObject);
```

Comment:
I think we should add a try catch here, and return an error message if its an invalid json

Reply:
You can try to convert it to the correct JSON object before calling `setAccountObject`